core/src/next_leader.rs

```diff
@@ -1,10 +1,40 @@
 use {
+    itertools::Itertools,
     solana_gossip::{cluster_info::ClusterInfo, contact_info::ContactInfo},
     solana_poh::poh_recorder::PohRecorder,
     solana_sdk::{clock::FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET, pubkey::Pubkey},
     std::{net::SocketAddr, sync::RwLock},
 };
 
+/// Returns a list of tpu vote sockets for the leaders of the next N fanout
+/// slots. Leaders are deduped but the resulting list could have duplicate
+/// sockets if two different leaders share the same tpu vote socket.
+pub(crate) fn upcoming_leader_tpu_vote_sockets(
+    cluster_info: &ClusterInfo,
+    poh_recorder: &RwLock<PohRecorder>,
+    fanout_slots: usize,
+) -> Vec<SocketAddr> {
+    let upcoming_leaders = {
+        let mut upcoming_leaders = Vec::with_capacity(fanout_slots);
+        let poh_recorder = poh_recorder.read().unwrap();
+        for n_slots in 1..=fanout_slots {
+            upcoming_leaders.push(poh_recorder.leader_after_n_slots(n_slots as u64));
```

Comment:
`None` values could be filtered out earlier here, for example:
```rust
upcoming_leaders.extend((1..=fanout_slots).flat_map(...))
```
also `dedup`ing (instead of `unique`) here would probably be fine.

Reply:
any reason not to use `filter_map` here over `flat_map`

Reply:
yeah i guess both would work